PoGo.NecroBot.Logic/Tasks/SnipePokemonTask.cs

```diff
@@ -79,6 +79,14 @@ public static class SnipePokemonTask
                     {
                         foreach (var pokemonLocation in locationsToSnipe)
                         {
+                            var pokeBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemPokeBall);
+                            var greatBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemGreatBall);
+                            var ultraBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemUltraBall);
+                            var masterBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemMasterBall);
+
+                            if (pokeBallsCount + greatBallsCount + ultraBallsCount + masterBallsCount < session.LogicSettings.MinPokeballsToSnipe)
+                                return;
+
```

Comment:
That's the problem though, you don't check other locations beyond the first 1-2 with this. It farms balls, gets just a few more than the limit to snipe, starts sniping at the first location, falls below the limit, stops sniping, repeat. It always starts sniping at the same location.